pytorch_lightning/trainer/trainer.py

```diff
@@ -1306,8 +1306,7 @@ def test(
 
     def __test_using_best_weights(self, ckpt_path, test_dataloaders):
         model = self.get_model()
-        if self.is_function_implemented('setup', model):
-            model.setup('test')
+        model.setup('test')
```

Comment:
Quick question: As of now, it is calling `trainer.setup('test')` and `model.setup('test')` outside mp process only once for ddp or tpu (8 cores) training or horovod. Shouldn't the testing setup be called within mp process too to ensure ddp or any multi-device testing?

Reply:
setup for anything DDP should happen inside spawn

Reply:
should I move it there then for all?

Reply:
@awaelchli Should I make the changes here too??

Reply:
The way you have it now looks good to me.